lib/AST/FrontendSourceFileDepGraphFactory.cpp

```diff
@@ -463,16 +470,28 @@ class UsedDeclEnumerator {
   void enumerateNominalUses(UseEnumerator enumerator) {
     auto &Ctx = SF->getASTContext();
     Ctx.evaluator.enumerateReferencesInFile(SF, [&](const auto &ref) {
-      const NominalTypeDecl *subject = ref.subject;
+      const DeclContext *subject = ref.subject;
       if (!subject) {
         return;
       }
 
-      auto key =
+      auto nominalKey =
           DependencyKey::Builder(NodeKind::nominal, DeclAspect::interface)
-              .withContext(subject)
+              .withContext(subject->getSelfNominalTypeDecl())
               .build();
-      enumerateUse(key, enumerator);
+      enumerateUse(nominalKey, enumerator);
+
+      auto declKey =
+          DependencyKey::Builder(NodeKind::nominal, DeclAspect::interface)
+              .withContext(subject->getAsDecl())
+              .build();
+
+      // If the subject of this dependency is not the nominal type itself,
+      // record another arc for the extension this member came from.
```

Comment:
This comment needs another sentence or two explaining the "why".

Reply:
Also, it ought to explain why just the interface and not the implementation for the sake of future maintainers.